pkg/test/framework/components/istio/util.go

```diff
@@ -58,3 +67,61 @@ func waitForValidationWebhook(accessor *kube.Accessor, cfg Config) error {
 		return fmt.Errorf("validation webhook not ready yet: %v", err)
 	}, retry.Timeout(time.Minute))
 }
+
+// TODO(landow) extract this "get nodeport or loadbalancer address" logic
+func getIstiodAddress(env *kubeenv.Environment, cluster kubeenv.Cluster) (net.TCPAddr, error) {
+	// In KinD, we don't have LoadBalancer support. Instead we do a little bit of trickery to to get the Node
+	// port for istiod though the ingressgateway service. The Minikube flag is a misnomer.
+	if env.Settings().Minikube {
+		pods, err := cluster.GetPods(ns, "istio=ingressgateway")
```

Comment:
If things are looking good I'll DRY this up since we do something similar in "components/ingress"